src/Symfony/Component/Serializer/NameConverter/MetadataAwareNameConverter.php

```diff
@@ -117,22 +121,22 @@ private function getCacheValueForAttributesMetadata(string $class, array $contex
 
         $classMetadata = $this->metadataFactory->getMetadataFor($class);
 
+        $contextGroups = (array) ($context[AbstractNormalizer::GROUPS] ?? []);
```

Comment:
You might follow https://github.com/symfony/symfony/pull/58705, as it might change this part of the code.